### PR TITLE
koordlet: add service monitor

### PR DESCRIPTION
**Pull request by fengyehong**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] I have bumped the chart version according to [versioning](https://github.com/koordinator-sh/charts/blob/master/CONTRIBUTING.md#versioning)
* [ ] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/koordinator-sh/charts/blob/master/CONTRIBUTING.md#changelog).
* [ ] Any new values are backwards compatible and/or have sensible default.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/koordinator-sh/koordinator/blob/master/CODE_OF_CONDUCT.md).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

File: versions/v1.0.0/templates/koordlet-service.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    koord-app: koordlet
+  name: koordlet
```
Comment (songtao98) [inline versions/v1.0.0/templates/koordlet-service.yaml:6]: consider use koordlet-service or koordlet-metrics-service instead

File: versions/v1.0.0/templates/koordlet-servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    koord-app: koordlet
+  name: koordlet
```
Comment (songtao98) [inline versions/v1.0.0/templates/koordlet-servicemonitor.yaml:6]: consider use koodlet-service-monitor

File: versions/v1.0.0/templates/koordlet-servicemonitor.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  endpoints:
+  - interval: 30s
+    port: http
+    relabelings:
+    - action: replace
+      regex: (.*)
+      replacement: $1
+      sourceLabels:
+      - __meta_kubernetes_pod_node_name
+      targetLabel: node
```
Comment (songtao98) [inline versions/v1.0.0/templates/koordlet-servicemonitor.yaml:18]: I think label `node` is already added in source code of koordlet, do we need to use `relabelings` to do this?

File: versions/v1.0.0/templates/koordlet-service.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
```
[the remark this answers is not in the document]
Comment (songtao98) [reply versions/v1.0.0/templates/koordlet-service.yaml:1]: https://github.com/koordinator-sh/charts/pull/54

File: versions/v1.0.0/templates/koordlet-servicemonitor.yaml (hunk 3 above)
Comment (fengyehong) [reply versions/v1.0.0/templates/koordlet-servicemonitor.yaml]: Indeed.

In servermonitor we can generate node label for all metrics with just a few lines, I think we can just remove calling and checking `genNodeLabels` in every metric in sourcecode. This is more efficient.

**Comment by zwzhang0107** [comment]
please add an option in value to control whether to install ServiceMonitor, since this CRD is imported by prometheus, which may not installed for some users.

**Comment by songtao98** [comment]
I've made a commit for the install control but received a permission denied when pushed. Maybe you can check `Allow edits from maintainers` or modify it by yourself? @fengyehong

**Pull request closed without merging**